### ISSUE TITLE
Add more control over which quotes - `string-quotes`

### ISSUE THREAD

**Issue by MadLittleMods** [stylelint/stylelint#476]
[`string-quotes`](https://github.com/stylelint/stylelint/blob/master/src/rules/string-quotes/README.md)

Enforce quotes differently in certain situations. For example attribute selectors remind me of the actual markup so I prefer to use double quotes but I still like single quotes everywhere else

```json
"string-quotes": [2, "single", {
  "attribute-selector": "double",
  "value": "single" <--- optional in this case
}]
```

```css
a[id="foo"] {
  content: 'x';
}
```

**Comment by jeddy3** [comment]
We originally had this discussion back in https://github.com/stylelint/stylelint/issues/96.

We concluded that "it's very reasonable to assume that users will want a consistent quote style wherever there are quotes". It did seem like a reasonable assumption, but here we are :)

There were a couple of issues raised in #96 about what level of granular control we should go for:

> David: I am not sure that distinguishing selector attribute quotes from all other "string" quotes makes a ton of sense unless we get even more granular: if the principle is that we're going to allow the user to specify a quote style in every location in which a string can be used, then it would be consistent; if not, isn't it a kind of arbitrary split to single out selector attributes?

> Maybe we could balance granularity and say "selector-quotes" vs. "value-quotes" ... but is that the whole picture?

So, a couple of points to decided on:

1. What is the correct level of granular control?
2. Is this functionality a good candidate for a plugin?
3. Do we use the "except" pattern e.g.

```js
"string-quotes": [2, "single", {
  "except": ["selector", "value"]
}]
```

**Comment by davidtheclark** [comment]
> Enforce quotes differently in certain situations.

I think this would cause unnecessary confusion without contributing additional value. It's going to take quite a bit of convincing before I dedicate time to writing code for this.

**Comment by jeddy3** [comment]
Agreed. Closing in favour of this edge case being handled outside of core and in a plugin.
